lib/nostrum/bot.ex

```diff
@@ -59,14 +59,27 @@ defmodule Nostrum.Bot do
     def handle_event(_), do: :ok
   ```
 
-  ## Limitations
-
-  This module was introduced in version `0.11.0` to provide a first step
-  towards migrating away from nostrum starting as an application. However, at
-  present, you are not able to start multiple bots with this at the same time
-  due to global names used internally. However, as we migrate away from the
-  application-based deployment, further internal changes should not require any
-  update from you.
+  ## Multiple Bots
+
+  Nostrum allows you to start multiple bots under your supervision tree. If you are
+  calling a function that interacts with an API specific to a bot - Rest, Cache, Voice, etc. -
+  nostrum needs to determine which bot to call it for. It does this with a bot `Registry`
+  and process dictionaries so that you don't have to pass an additional argument to every
+  function.
+
+  Inside the `handle_event/1` callback in your consumer, the calling process already
+  has the context of the bot that generated the event for any API calls that follow.
+
+  If you have a single bot running and need to call an API asynchronously, i.e. *not* as a response
+  to a gateway event in your consumer, nostrum will automatically determine the bot.
+
+  However, if you have multiple bots running *and* you need to make unprompted bot calls, you
+  will need to assist nostrum by setting the process's context:
+  - `use Nostrum.Bot, name: MyBot` at the top of the module where the API-calling functions are defined, or
+  - `Nostrum.Bot.set_bot_name(MyBot)` dynamically before each API call
```

Comment:
What do you think about also having a function similar to `Nostrum.Bot.with_bot(MyBot, fn -> ... end), which sets the bot to use and then clears it / sets it back to what it was before returning?

I'm specifically thinking something similar to how `Ecto.Repo.transaction` works.

To me that'd be easier to understand/reason about than having `use Nostrum.Bot, name: MyBot`. Its less difficult to make sense of what its doing than us patching `def` the way we do.